### ISSUE TITLE
EXCEPTION: Promise is not a constructor in v12.0.0-rc5

### ISSUE THREAD

**Issue by georgeedwards** [elastic/elasticsearch-js#449]
I am trying to get this working with an angular2 application. I am importing this (elasticsearch.js) into my component and then:

```
var connectionString = 'xyz';
this._client = new elasticsearch.Client({ host: connectionString, log: 'trace' });

ngOnInit() {
    console.log('NGINIT');
    /*this._client.search({
      index: 'twitter',
      type: 'tweets',
      body: {
        query: {
          match: {
            body: 'elasticsearch'
          }
        }
      }
    }).then(function (resp) {
      var hits = resp.hits.hits;
    }, function (err) {
      console.trace(err.message);
    });
*/

  }
```

so in my console I get:

> DEBUG: 2016-10-02T12:09:56Z
>   starting request {
>     "method": "POST",
>     "path": "/plugins/document/_search",
>     "body": {
>       "query": {
>         "query_string": {
>           "query": "Bluetooth"
>         }
>       }
>     },
>     "query": {}
>   }

and then when I uncomment the `this._client.search({` section, I get:

> EXCEPTION: Promise is not a constructor

```
client.ping({ ...})
```

works as expected. and the issue goes away if I use **v.11.0.1** rather than **v.12.0.0-rc5**

**Comment by kmulvey** [comment]
i have the same issue, seems to be coming from:
im using version 12.1.2

```
	Transport.prototype.defer = function () {
	  var defer = {};
	  defer.promise = new Promise(function (resolve, reject) {
	    defer.resolve = resolve;
	    defer.reject = reject;
	  });
	  return defer;
	};
```

**Comment by spalger** [comment]
can you please describe how you are importing/consuming elasticsearch.js? Are you using npm + webpack? Script tag from a CDN?

**Comment by kmulvey** [comment]
<script src="bower_components/elasticsearch/elasticsearch.js"></script>

chrome Version 56.0.2924.21 beta (64-bit)

**Comment by spalger** [comment]
Apologies, I accidentally ignored the promise polyfill in the vanilla browser build. Fixed in `12.1.3`, which I just published.

**Comment by kmulvey** [comment]
totally works now, thanks
